CMakeLists.txt

```diff
@@ -3,6 +3,17 @@
 cmake_minimum_required (VERSION 3.5)
 PROJECT(toolkitICL)
 
+ find_package(CUDA)
+ 
+IF( CUDA_FOUND)
+ option(USENVML "Use NVML" ON)
```

Comment:
Maybe it would be better to use something like
```
IF(CUDA_FOUND)
  option(USENVML "Use NVML" ON)
ELSE()
  option(USENVML "Use NVML" OFF)
ENDIF()

IF(USENVML)
  MESSAGE(STATUS "Using NVML")
  add_definitions(-DUSENVML)
ENDIF()
```

Reply:
Using Windows, this overrides 'cmake -DUSENVML=OFF' if CUDA is detected. I'm guessing this is the intention behind it?

Reply:
You're right We (you?) should decide whether `USENVML` shall be an option for the user or not. If it is no option, we can use something as above. Otherwise, an additional check has to be added to the first `IF`. But I think it might be nice to turn `USENVML=OFF` if CUDA isn't found (with an additional warning, maybe?).